crates/solidity/outputs/npm/package/src/generated/index.d.ts

```diff
@@ -507,30 +507,30 @@ export namespace cst {
   }
   export class RuleNode {
     get type(): NodeType.Rule;
-    get kind(): RuleKind;
-    get textLength(): TextIndex;
-    get children(): Array<RuleNode | TokenNode>;
-    get cursor(): Cursor;
+    get kind(): kinds.RuleKind;
+    get textLength(): text_index.TextIndex;
+    get children(): Array<cst.RuleNode | cst.TokenNode>;
+    get cursor(): cursor.Cursor;
   }
   export class TokenNode {
     get type(): NodeType.Token;
-    get kind(): TokenKind;
-    get textLength(): TextIndex;
+    get kind(): kinds.TokenKind;
+    get textLength(): text_index.TextIndex;
     get text(): string;
-    get cursor(): Cursor;
+    get cursor(): cursor.Cursor;
```

Comment:
I suggest making APIs like `.cursor` and similar ones functions instead of getters. It creates a new cursor every time, rather than returning some data (property) of the object (node) that the property belongs to. It also hints that the cursor returned should be persisted/reused, and can mutate and change independently of the node.